common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsEvent.kt

```diff
@@ -0,0 +1,93 @@
+package com.revenuecat.purchases.common.diagnostics
+
+import com.revenuecat.purchases.common.DateProvider
+import com.revenuecat.purchases.common.DefaultDateProvider
+import org.json.JSONArray
+import org.json.JSONObject
+import java.util.Date
+
+sealed class DiagnosticsEvent(val diagnosticType: String) {
+    companion object {
+        private const val VERSION_KEY = "version"
+        private const val TYPE_KEY = "type"
+
+        private const val VERSION = 1
+    }
+
+    data class Log(
+        val name: DiagnosticsLogEventName,
+        val properties: Map<String, Any>,
+        val dateProvider: DateProvider = DefaultDateProvider(),
+        val time: Date = dateProvider.now
```

Comment:
I used this mechanism to automatically add the timestamp which we use in `SubscriberAttributes`.

Reply:
I would rename this to `dateTime`, so time is accessed `dateTime.time`